src/_rules/layout.js

```diff
@@ -17,3 +19,18 @@ export const overflows = [
       (overflowValues.includes(v) ? { [`overflow-${d}`]: v } : undefined),
   ],
 ];
+
+export const columns = [
+  [
+    /^columns-(\d+)$/,
+    ([, d]) => ({ 'columns': h.number(d) }),
+    { autocomplete: 'columns-<num>' },
+  ],
+  ['columns-auto', { 'columns': 'auto' }],
+  // matching arbitrary values
+  [
+    /^columns-\[(.\d*)?]$/,
```

Comment:
Should this be something like 
```suggestion
    /^columns-\[(\d\d*)?]$/,
```
to not accept things like `columns-[z33]`? 🤔